src/core/net/srp_server.cpp

```diff
@@ -855,15 +855,15 @@ Error Server::ProcessServiceDescriptionInstructions(Host &                   aHo
     // that `mTimeLastUpdate` on a new `Service::Description` is set to
     // `GetNow().GetDistantPast()`.
 
-    for (desc = aHost.mServiceDescriptions.GetHead(); desc != nullptr; desc = desc->GetNext())
+    for (Service::Description &dsc : aHost.mServiceDescriptions)
```

Comment:
nit:
```suggestion
    for (Service::Description &desc : aHost.mServiceDescriptions)
```

Reply:
We already have `desc` variable in the scope above in this method, so if we re-use the same name in the loop, we get a warning/error that the loop variable is shadowing an existing variable name.

```
src/core/net/srp_server.cpp:858:32: error: declaration shadows a local variable [-Werror,-Wshadow]
    for (Service::Description &desc : aHost.mServiceDescriptions)
```

Reply:
oh, I missed this fact. But it looks more like a typo to name it `dsc` :(

What about moving the `Service::Description *desc;` into the `for` loop above? As that `desc` is never referenced out of that loop.

Reply:
Yes. I think that can work. Good idea. Will change in next push.

Reply:
Thanks!